realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1344,6 +1276,10 @@ public void run() {
                                 });
                             }
                         }
+
+                        if (!handlerController.threadContainsAsyncQueries()) {
+                            handlerController.executeAsyncTransactionCallbacks();
```

Comment:
I would like to simplify the logic by just calling `executeAsyncTransactionCallbacks` in `HandlerController.realmChanged()` the else clause there.